mtags/src/main/scala/scala/meta/internal/metals/BinaryNameDecoder.scala

```diff
@@ -0,0 +1,60 @@
+package scala.meta.internal.metals
+
+import scala.annotation.tailrec
+
+object BinaryNameDecoder {
```

Comment:
Can we use compiler defined `NameTransfoer`'s decode methods?
- scala2: https://github.com/scala/scala/blob/7952071b88a85e2502e9ecd4e8bd57040d9448f5/src/library/scala/reflect/NameTransformer.scala#L98-L103
- scala3: https://github.com/lampepfl/dotty/blob/31be2b0843d6bd9317c0ef9f2af9d8ec93271a55/compiler/src/dotty/tools/dotc/util/NameTransformer.scala#L86-L89

Reply:
Nice catch!
I forgot that server has a compiler dependency